File: CS/RestoreLayoutWithOverlappedOptions/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using System.Collections;
using System.IO;

namespace RestoreLayoutWithOverlappedOptions {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void btnRestore_Click(object sender, EventArgs e) {
            
            string modifiedLayout = cleanupReportLayout(File.ReadAllText("TestLayout.repx"));
            File.WriteAllText("ModifiedLayout.repx", modifiedLayout);
            XtraReport report = XtraReport.FromFile("ModifiedLayout.repx", true);
            report.ShowPreview();
        }
        private string cleanupReportLayout(string ReportLayout) {
            if (!string.IsNullOrEmpty(ReportLayout)) {

                // Cleanup Report Layout if Required

                if (ReportLayout.Contains("OverlappingOptions")) {
                    ArrayList controlNames = new ArrayList();
                    using (StringWriter sw = new StringWriter()) {
                        using (StringReader sr = new StringReader(ReportLayout)) {

                            string line;
                            while ((line = sr.ReadLine()) != null) {
                                if (!line.Contains("OverlappingOptions")) {
                                    bool bFound = false;
                                    foreach (string controlName in controlNames) {
                                        if (line.Contains(controlName)) {
                                            bFound = true;
                                            break;
                                        }
                                    } 
                                    if (!bFound) {
                                        if (!line.Contains("OverlappingOptionsTypeName")) {
                                            sw.WriteLine(line);
                                        } 
                                    } 
                                }
                                else {
                                    char[] splitchar = new char[1];
                                    splitchar[0] = ' ';
                                    string[] parms = line.Trim().Split(splitchar);
                                    controlNames.Add(parms[0]);
                                } 
                            } 
                        }
                        ReportLayout = sw.ToString();
                    } 
                } 
            } 
            return ReportLayout;
        }
    }
}

File: VB/RestoreLayoutWithOverlappedOptions/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports System.Collections
Imports System.IO

Namespace RestoreLayoutWithOverlappedOptions
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRestore_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRestore.Click

			Dim modifiedLayout As String = cleanupReportLayout(File.ReadAllText("TestLayout.repx"))
			File.WriteAllText("ModifiedLayout.repx", modifiedLayout)
			Dim report As XtraReport = XtraReport.FromFile("ModifiedLayout.repx", True)
			report.ShowPreview()
		End Sub
		Private Function cleanupReportLayout(ByVal ReportLayout As String) As String
			If (Not String.IsNullOrEmpty(ReportLayout)) Then

				' Cleanup Report Layout if Required

				If ReportLayout.Contains("OverlappingOptions") Then
					Dim controlNames As New ArrayList()
					Using sw As New StringWriter()
						Using sr As New StringReader(ReportLayout)

							Dim line As String
							line = sr.ReadLine()
							Do While line IsNot Nothing
								If (Not line.Contains("OverlappingOptions")) Then
									Dim bFound As Boolean = False
									For Each controlName As String In controlNames
										If line.Contains(controlName) Then
											bFound = True
											Exit For
										End If
									Next controlName
									If (Not bFound) Then
										If (Not line.Contains("OverlappingOptionsTypeName")) Then
											sw.WriteLine(line)
										End If
									End If
								Else
									Dim splitchar(0) As Char
									splitchar(0) = " "c
									Dim parms() As String = line.Trim().Split(splitchar)
									controlNames.Add(parms(0))
								End If
								line = sr.ReadLine()
							Loop
						End Using
						ReportLayout = sw.ToString()
					End Using
				End If
			End If
			Return ReportLayout
		End Function
	End Class
End Namespace